com.unity.render-pipelines.high-definition/Editor/Material/UIBlocks/HDShaderGUI.cs

```diff
@@ -34,6 +45,11 @@ public override void AssignNewShaderToMaterial(Material material, Shader oldShad
             SetupMaterialKeywordsAndPassInternal(material);
         }
 
+        /// <summary>
+        /// Setup the keywords and passes for the material. It is required to call this function after changing a property on a material to ensure it's validity.
```

Comment:
Sets up the keywords and passes for the material. You must call this function after you change a property on a material to ensure it's validity.